pkg/config/config.go

```diff
@@ -1,201 +1,274 @@
+		return err
+	}
+	c.path = path
+	return nil
 }
 
-// SetReporting sets whether to report usage events to Mesosphere.
-func (conf *Config) SetReporting(reporting bool) {
-	conf.reporting = reporting
-	conf.setDirty(&conf.reporting)
+// LoadReader populates the store based on an io.Reader containing TOML data.
+func (c *Config) LoadReader(reader io.Reader) error {
+	tree, err := toml.LoadReader(reader)
+	if err != nil {
+		return err
+	}
+	c.LoadTree(tree)
+	return nil
 }
 
-// MesosMasterURL returns the Mesos master URL. Must be set in format "http://host:port".
-func (conf *Config) MesosMasterURL() string {
-	return conf.mesosMasterURL
+// LoadTree populates the store with a TOML tree.
+func (c *Config) LoadTree(tree *toml.Tree) {
+	c.tree = tree
 }
 
-// SetMesosMasterURL sets the Mesos master URL. Must be set in format "http://host:port".
-func (conf *Config) SetMesosMasterURL(mesosMasterURL string) {
-	conf.mesosMasterURL = mesosMasterURL
-	conf.setDirty(&conf.mesosMasterURL)
+// Path returns the path to the Config.
+func (c *Config) Path() string {
+	return c.path
 }
 
-// PromptLogin returns whether to automatically prompt the user to log in when their token expired.
-func (conf *Config) PromptLogin() bool {
-	return conf.promptLogin
+// SetPath assigns a path to the Config.
+func (c *Config) SetPath(path string) {
+	c.path = path
 }
 
-// SetPromptLogin sets whether to automatically prompt the user to log in when their token expired.
-func (conf *Config) SetPromptLogin(promptLogin bool) {
-	conf.promptLogin = promptLogin
-	conf.setDirty(&conf.promptLogin)
+// Fs returns the filesystem for the store.
+func (c *Config) Fs() afero.Fs {
+	return c.fs
 }
 
-// ClusterName returns the custom name for the cluster.
-func (conf *Config) ClusterName() string {
-	return conf.clusterName
+// Attached returns whether or not the store has an attached file next to it.
+func (c *Config) Attached() bool {
+	attachedFilePath := filepath.Join(filepath.Dir(c.path), "attached")
+	attachedFileInfo, err := c.fs.Stat(attachedFilePath)
+	if err == nil && attachedFileInfo.Mode().IsRegular() {
+		return true
+	}
+	return false
 }
 
-// SetClusterName sets a custom name for the cluster.
-func (conf *Config) SetClusterName(clusterName string) {
-	conf.clusterName = clusterName
-	conf.setDirty(&conf.clusterName)
-}
+// Get returns a value from the Config using a key.
+func (c *Config) Get(key string) interface{} {
+	// Check if the given key is whitelisted as an env var.
+	if envVar, ok := c.envWhitelist[key]; ok {
+		// If so, look-it up.
+		if envVal, ok := c.envLookup(envVar); ok {
+			return envVal
+		}
+	}
 
-// setDirty sets a field as "dirty", this is used by the Marshal
-// function to know which fields must be persisted to the Store.
-func (conf *Config) setDirty(field interface{}) {
-	conf.dirtyFields[field] = true
+	// Fallback to the TOML tree if present.
+	switch node := c.tree.Get(key).(type) {
+	case *toml.Tree, []*toml.Tree:
```

Comment:
I don't see how could the value returned be of type `[]*toml.Tree` with https://github.com/pelletier/go-toml/blob/66540cf1fcd2c3aee6f6787dfa32a6ae9a870f12/toml.go#L78 🤔

Reply:
This happens with tables (cf. https://github.com/toml-lang/toml#array-of-tables). We don't support it in our CLI configuration so here I'd rather return `nil`.